dune_client/api/base.py

```diff
@@ -179,15 +179,23 @@ def _get(
             return response
         return self._handle_response(response)
 
-    def _post(self, route: str, params: Optional[Any] = None) -> Any:
+    def _post(
+        self,
+        route: str,
+        params: Optional[Any] = None,
+        data: Optional[IO[bytes]] = None,
+        headers: Dict[str, str] = {},
+    ) -> Any:
+        # pylint: disable=dangerous-default-value
         """Generic interface for the POST method of a Dune API request"""
         url = self._route_url(route)
         self.logger.debug(f"POST received input url={url}, params={params}")
         response = self.http.post(
             url=url,
             json=params,
-            headers=self.default_headers(),
+            headers=dict(self.default_headers(), **headers),
             timeout=self.request_timeout,
+            data=data,
```

Comment:
Are we streaming the data here ?

According to the [docs](https://requests.readthedocs.io/en/latest/user/advanced/#streaming-uploads): "_To stream and upload, simply provide a file-like object for your body_". However I am not sure if IO is a [file-like object](https://docs.python.org/3/glossary.html#term-file-like-object).

edit. Based on the way you are using the function on tests, looks like we are streaming. Just wanted to draw your attention.

Reply:
So are you recommending a different type of `typing` object?

Reply:
Not sure, this question stems from my inexperience using Python. The more I read about [it](https://docs.python.org/3/library/typing.html#typing.IO), `typing.IO` looks like what we need. We need to be sure that we don't load the whole file in memory before posting to the server 😄

Reply:
Similar to what you shared, I believe the `BufferedReader`, which is what the user would be using is represented in `typing.IO`
https://stackoverflow.com/questions/38569401/type-hint-for-a-file-or-file-like-object